common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/OneForOneBlockFetcher.java

```diff
@@ -151,14 +167,21 @@ private void failRemainingBlocks(String[] failedBlockIds, Throwable e) {
     }
   }
 
+  public void cleanup() {
+    for (File file: shuffleFiles) {
+      file.delete();
+    }
+  }
+
   private class DownloadCallback implements StreamCallback {
 
     private WritableByteChannel channel = null;
     private File targetFile = null;
     private int chunkIndex;
 
-    DownloadCallback(File targetFile, int chunkIndex) throws IOException {
-      this.targetFile = targetFile;
+    DownloadCallback(int chunkIndex) throws IOException {
+      this.targetFile = tmpFileCreater.createTempBlock();
+      shuffleFiles.add(targetFile);
```

Comment:
ditto, we will create unique temp files, so `shuffleFiles` can be a list instead of set.